### DISCUSSION TITLE
requirements.txt

### DISCUSSION THREAD

**Discussion by RaivoKoot** [pykale/pykale#20] [General] [answered]
To show people what python dependencies need to be installed to use pykale, we have a requirements.txt in the root folder.
We will need to revise this at some point and I'm making this `issue` as a reminder for us.  
  
Currently, different parts of the library require different dependencies. The new video loading module I added requires at least Python 3.7, at least torch 1.7.0 and at least torchvision 0.8.0. These are very new versions and I don't know whether overwriting the current version requirements for these packages inside our `requirements.txt` will give problems to people that can't use these newer versions yet.

We will have many different modules, each with very different dependencies, which altogether form many dependencies. We'll have to think and see at some point how exactly we want to handle this and our `requirements.txt`.

**Comment by haipinglu** [comment, marked as answer]
Yes, this is an important issue and it should have been covered in David's
recommendations as well at https://github.com/pykale/pykale/projects/2
We will need a discussion on it with David and the team. I will set up a
meeting later. Many thanks.

Kind regards,

Haiping @ http://staffwww.dcs.shef.ac.uk/people/H.Lu/
Senior Lecturer in Machine Learning, University of Sheffield

**Comment by RaivoKoot** [comment]
Nice, that sounds good.
